### ISSUE TITLE
Significant slowdown on multiple stages

### ISSUE THREAD

**Issue by DracoOmega** [Kayinnasaki/BepIssueTracker#27]
I felt a bit hesitant to post this as an 'issue', since I know you're already aware of it, and maybe can't even do much to address it, given the tools the game is built with, but I felt it perhaps bad policy to stay silent on an issue that is (sadly) significantly impacting enjoyment of an otherwise great game.

There are very significant quantities of slowdown in a number of stages. I first noticed it being kind of bad for a bunch of chapter 3, even from early in that stage (maybe moreso in Naomi's route than Sinlin's?). But all of Naomi chapter 4 was fairly slow, and her chapter 5 was even worse (I might estimate that the game spent the majority of time running close to half-speed? Which sadly gets extremely wearying when played in slow-mo for an extended length of time)

I did notice, once or twice in Sinlin's lower path in chapter 3, that a seed pod from plants on the route far above would drift down (rarely enough that it looked like not an intended hazard?). I wonder if this means that large areas of off-screen level are being simulated and rendered, which might be a major factor in this slowdown? Maybe there's some way to stop Construct from doing this? (I also wondered if the animated rain might be part of the slowdown in some areas?). I don't actually know what can be done about this, sadly - but I felt I should point out how strongly I've been affected by it.

My computer is a few years old now, but was on the higher-end when I bought it, and it continues to run relatively demanding games without much issue (I can provide specs if they'd be at all helpful?)

**Comment by Kayinnasaki** [comment]
Yeah, this problem is huge motherfucker. There are some issues that just cannot be solved with how rending works (due to the 'zoomed in' nature of the game, the game renders far more than is actually needed because, well, construct is dumb).

Where you playing full screen/large window? If so I suggest playing at the default 2x and seeing how much that helps.

**Comment by DracoOmega** [comment]
It was with default window size, I'm afraid.

And is it really literally impossible in Construct to tell it not to render various off-screen stuff? Like, I'm not actually disagreeing with you (I know virtually nothing of Construct), but the programmer in me is frowning at this lack of functionality. I know the original Spelunky did some magic to not simulate off-screen things for performance reasons (even if this caused some mechanical oddities). Of course, that was Game Maker and not Construct, so kind of irrelevant here.

**Comment by Kayinnasaki** [comment]
I dunno how up to date the last version of the game was (I went in to do some performance tweaks at some point) but if you run the launcher there is a new version. So if you want you can try stage 3 and 5 and tell me if they at least run better.

**Comment by DracoOmega** [comment]
I tried stage 3 and 5 again. Stage 3's performance might be slightly improved (though it was somewhat hard for me to be sure if it didn't just seem better in comparison to more recent experience with 5), but I'm afraid I didn't notice any improvement in stage 5. Moreover, when I progressed further in stage 5 than I had done previously, it got substantially worse.

The point where all the skeletons clamour out of the pit caused the game to slow to a crawl (and persisted this way even into and during the boss fight). Dying to her did at least revert it to its previous level of slowness, which was more manageable by comparison (though still unpleasant).

One (probably useless) curiosity is that exactly ONCE, the boss fight and walk back to the boss after dying ran at a normal speed. I had alt-tabbed at some point (maybe while dying?) and when I had returned, performance was very noticeably improved. And it STAYED improved for another attempt, at which point it all reverted back to how it was before. I could not reproduce this (though I am quite sure it was not due a change in other load on my cpu, since I wasn't doing any demanding tasks either before or after). It sort of vaguely implies to me that either this caused something to glitch out and not load, or that there's SOMETHING more subtle and possibly avoidable code-wise that is causing additional lag? (Though the skeleton pit shows pretty clearly that some large part of if it from the number of active entities)

On the plus side, I found that raising the thread priority of the game in task manager does help a little. Doesn't alleviate it completely, but it's at least some modest improvement.
